### ISSUE TITLE
Trying to soft-delete an existing stream with StreamState.Exists throws

### ISSUE THREAD

**Issue by gingters** [EventStore/EventStore#3151]
**Describe the bug**

If trying to soft-delete a stream with StreamState.Exists it throws an exception even though the stream exists.
When using StreamState.Any it works.

**To Reproduce**
I started with creating a minimal sample client:
1. `dotnet new console`
2. `dotnet add package EventStore.Client.Grpc.Streams` 
3. Add this Program.cs
```C#
using System;
using System.Linq;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;
using EventStore.Client;

namespace EventStoreTestClient
{
	public class Program
	{
		public static async Task<int> Main(string [] args)
		{
			Console.WriteLine(".NET 5 sample, using EventStore.Client.Grpc.Streams 21.2.0");

			var client = new EventStoreClient(EventStoreClientSettings.Create("esdb://localhost:2113?tls=false"));
		
			// make sure a new stream exists
			var evt = new EventData(Uuid.NewUuid(), "StreamCreated", JsonSerializer.SerializeToUtf8Bytes(new { Id = Guid.NewGuid(), SomeData = "Some Test", }));
			await client.AppendToStreamAsync("test-stream", StreamState.NoStream, new [] { evt, });

			try
			{
				// and try to soft-delete that existing stream
				var res = await client.SoftDeleteAsync("test-stream", StreamState.StreamExists);
			}
			catch (Exception ex)
			{
				Console.WriteLine(ex);
			}

			return 0;
		}
	}
}
```
4. Run the sample

**Expected behavior**
The existing stream should be soft-deleted.

**Actual behavior**
I get an exception with this details:

```
System.InvalidOperationException: Status(StatusCode="Unknown", Detail="Exception was thrown by handler.")
 ---> Grpc.Core.RpcException: Status(StatusCode="Unknown", Detail="Exception was thrown by handler.")
   --- End of inner exception stack trace ---
   at EventStore.Client.Interceptors.TypedExceptionInterceptor.<AsyncUnaryCall>b__6_0[TRequest,TResponse](Task`1 t)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__277_0(Object obj)
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at EventStore.Client.EventStoreClient.DeleteInternal(DeleteReq request, EventStoreClientOperationOptions operationOptions, UserCredentials userCredentials, CancellationToken cancellationToken)
   at EventStoreTestClient.Program.Main(String[] args) in C:\Dev\tt\research\event_sourced_idp\src\samples\EventStoreTestClient\Program.cs:line 29
```

**EventStore details**
 - EventStore server version: 21.6.0 running in a local docker container

 - Operating system: Windows 10

 - EventStore client version (if applicable): 21.2.0

**Comment by oskardudycz** [comment]
Looks like a database bug related to the condition: https://github.com/EventStore/EventStore/blame/master/src/EventStore.Core/Messages/ClientMessage.cs#L467.

**Comment by oskardudycz** [comment]
@gingters, the fix should be available in the upcoming 21.10 release 👍

**Comment by gingters** [comment]
Awesome, thanks!
